static/templates/message_edit_form.hbs

```diff
@@ -8,13 +8,15 @@
     {{#if is_stream}}
     <div class="edit-controls">
         <div class="message_edit_header">
-            <div class="stream_header_colorblock" {{#unless is_stream_editable}}style="display:none"{{/unless}}></div>
-            <div class="select_stream_setting" {{#unless is_stream_editable}}style="display:none"{{/unless}}>
-                {{> settings/dropdown_list_widget
-                  widget_name=select_move_stream_widget_name
-                  list_placeholder=(t 'Filter streams')}}
-            </div>
-            <i class="fa fa-angle-right" aria-hidden="true" {{#unless is_stream_editable}}style="display:none"{{/unless}}></i>
+            {{#if is_stream_editable}}
+                <div class="stream_header_colorblock"></div>
+                <div class="select_stream_setting">
+                    {{> settings/dropdown_list_widget
+                      widget_name=select_move_stream_widget_name
+                      list_placeholder=(t 'Filter streams')}}
+                </div>
+                <i class="fa fa-angle-right" aria-hidden="true"></i>
+            {{/if}}
```

Comment:
Just a note -- this sort of change isn't always an improvement, in that it makes live-update if the user's permissions change more difficult to support (where one could just show/hide the elements). 

I would check if we have existing such live-update logic that this change is breaking; it seems like the kind of thing we might have done.

Reply:
@timabbott , I confirmed it on local that we do not live-update this message-edit form on change of permission;

we need to refresh the page or else login again to see the edit-form with updated permissions;

Reply:
So I think it should be fine to go ahead with this change for now and can consider it as an improvement.

Reply:
Looking further, I'm pretty sure this change is confusing at best as written. In `message_edit.js`, we call `stream_bar.decorate(message.stream, $stream_header_colorblock, false);`, which will try to set styles for those elements.

So I'd prefer you rework this to preserve the model that those elements are in the DOM but not visible when we want them that way; it's better for live-update and will be easier to review, since it doesn't require analyzing the impact of that change.